### ISSUE TITLE
Fix up JSDoc type hinting 

### ISSUE THREAD

**Issue by chillu** [silverstripe/silverstripe-framework#5294]
We're following the AirBnB coding conventions (https://github.com/airbnb/javascript), who are implicitly using curly braces for types rather than just writing out the type name. This is pretty standard, see https://github.com/airbnb/javascripthttps://en.wikipedia.org/wiki/JSDoc.

Wrong

```
/**
 * @param Integer setid
 * @param object config
 */
```

Right

```
/**
 * @param {number} setid
 * @param {object} config
 */
```

See http://usejsdoc.org/tags-param.html and http://usejsdoc.org/tags-type.html

**Comment by sminnee** [comment]
Will this trigger an eslint error? If not, can we configure estlint to do so?

**Comment by chillu** [comment]
Not exactly, but we probably should turn on this option: http://eslint.org/docs/rules/valid-jsdoc#prefertype

**Comment by tractorcow** [comment]
If it's not an explicit error under the airBnB standard (even if they are using it as a matter of practice) this should be a warning not an error.

**Comment by sminnee** [comment]
+1 to stricter linting. It keeps things consistent. A warning would be fine but I'd probably still break the build on a warning tbh ;)
